encoder/src/main/java/openlr/encoder/routesearch/SecondShortestRouteChecker.java

```diff
@@ -0,0 +1,192 @@
+                    destination);
+
+            if (data.hasLengthValue(child)) {
+
+                if (data.getLengthValue(child) < distanceCoveredByChild) {
+                    PQElem newElem = new PQElem(child, heuristics, distanceCoveredByChild, parent);
+                    data.updateInOpen(newElem);
+                }
+            } else {
+                data.addToOpen(new PQElem(child, heuristics, distanceCoveredByChild, parent));
+
+            }
+        }
+    }
+
+    /**
+     * length of the alternate path
+     * @param destination road segment where the alternate path join back to the actual location
+     * @param index index of the road segment where alternate path starts
+     * @return length of the alternate bathe between start and route segments of alternate path.
+     */
+    private int lengthOfSecondShortestRoute(final PQElem destination, final int index) {
+        Long deviationStart = location.get(index - 1).getID();
+        int secondShortestRouteLength = 0;
+        PQElem to = destination;
+        while (to.getPrevious() != null && to.getPrevious().getLine().getID() != deviationStart) {
+            secondShortestRouteLength += to.getPrevious().getLine().getLineLength();
+            to = to.getPrevious();
+        }
+        return secondShortestRouteLength;
+    }
+
+    /**
+     *
+     * @param closedSet set of road segments which need not be explored
+     * @param routeSearchData priority queue
+     * @param index index of the road segment along the location where the second shortest route deviates from shortest route.
+     * @return true: if an alternate path with length less than the threshold exist
+     *         false: if no alternate path with length less than the threshold exist
+     */
+    private boolean exploreNetworkForSecondShortestRouteUnderThreshold(Set<Long> closedSet, RouteSearchData routeSearchData, int index) {
+        List<Long> possibleDestinations = location.stream().map(Line::getID).collect(Collectors.toList()).subList(index + 1, location.size());
+        while (!routeSearchData.isOpenEmpty()) {
+            PQElem parent = routeSearchData.pollElement();
+            if (possibleDestinations.contains(parent.getLine().getID())) {
+                int destinationIndexOnLocation = location.subList(index, location.size()).indexOf(parent.getLine());
+                int subLocationLength = location.subList(index, destinationIndexOnLocation + index).stream().mapToInt(Line::getLineLength).sum();
+                int secondShortestRouteLength = lengthOfSecondShortestRoute(parent, index);
+                return VerifyThreshold(secondShortestRouteLength, subLocationLength);
+            } else {
+                List<Line> children = getAcceptableSuccessors(parent, closedSet);
+                updateRouteSearchData(routeSearchData, children, parent);
+                closedSet.add(parent.getLine().getID());
+            }
+        }
+        return false;
```

Comment:
you are returning a boolean value, the the name of you method should indicate it. You can rename for example to: **IsNetworkForSecondShortestRouteUnderThreshold**

Reply:
ac5407dc43f6f89b768b92e9571ed9f04942a5d2